jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -98,8 +98,41 @@ public static boolean getDefaultUseCaches()
         return __defaultUseCaches;
     }
 
+    public static Resource.Mount mountCollection(Collection<Resource> resources)
+    {
+        List<Resource.Mount> mounts = new ArrayList<>();
+        resources.forEach(res ->
+        {
+            Resource.Mount mount = Resource.mountIfNeeded(res);
+            if (mount != null)
+                mounts.add(mount);
+        });
+
+        try
+        {
+            return new ResourceCollection.Mount(resources, mounts);
+        }
+        catch (Throwable t)
+        {
+            // can't create ResourceCollection.Mount, unmount and rethrow.
+            mounts.forEach(IO::close);
+            throw t;
+        }
+    }
+
+    public static Resource.Mount mountCollectionURIs(Collection<URI> uris)
+    {
+        return mountCollection(uris.stream().map(Resource::newResource).toList());
+    }
+
+    public static Resource.Mount mountCollection(String csvResources) throws IOException
+    {
+        return mountCollection(Resource.fromList(csvResources, false));
```

Comment:
This won't work as if one csv entry does need mounting, trying to create a `Resource` from it will throw.

Reply:
Completely reworked.

Now `Resource.split(String)` (simpler replacement for old-school `Resource.fromList()`) handles most of this by converting only to URI.
Letting other code manage the URI to Resource mapping.